### PR TITLE
Add opcua mapper

### PR DESCRIPTION
**Pull request by sailorvii**
The issue is at https://github.com/kubeedge/kubeedge/issues/2461.

This mapper uses open source code github.com/gopcua/opcua as the driver.
The code framework refers to the modbus mapper.
It has been tested with simulated opcua server & device: https://python-opcua.readthedocs.io/en/latest/.

### CODE REVIEW

**Comment by sailorvii** [comment]
@fisherxu @luogangyi 
Please help to review, thanks.

File: pkg/opcua/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM ubuntu:16.04
+
+RUN mkdir -p kubeedge ca
+
+COPY ./opcua kubeedge/
+COPY ./config.yaml kubeedge/
+COPY ./ca /ca
```
Comment (daixiang0) [inline pkg/opcua/Dockerfile:7]: It could be better if provide a tool to generate certs.

Comment (sailorvii) [reply pkg/opcua/Dockerfile:7]: Thanks for your comment. The cert is a demo. It depends on the external opcua server.

Comment (daixiang0) [reply pkg/opcua/Dockerfile:7]: I think we should provide a tool to generate them although it is a demo.

File: Makefile [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
-mappers:modbusmapper
+.PHONY: opcuamapper
+opcuamapper:
+	go build -o ./pkg/opcua/opcua ./pkg/opcua
+.PHONY: opcuamapper_image
+opcuamapper_image:opcuamapper
+	sudo docker build -t opcuamapper:v1.0 ./pkg/opcua
+
+clean:
+	rm -f ./pkg/modbus/modbus ./pkg/opcua/modbus
```
Comment (fisherxu) [inline Makefile:19]:
```suggestion
	rm -f ./pkg/modbus/modbus ./pkg/opcua/opcua
```
?

File: pkg/opcua/Dockerfile (hunk 1 above)
Comment (fisherxu) [reply pkg/opcua/Dockerfile:7]: We can remove it from dockerfile/image, and mount into contaier if needed

File: pkg/opcua/ca/pass [hunk 3]
```diff
@@ -0,0 +1 @@
+testpass
```
Comment (fisherxu) [inline pkg/opcua/ca/pass:1]: ?

File: pkg/opcua/driver/client_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		SecurityPolicy: "None",
+		SecurityMode:   "None",
+		*/
+		/*
+			SecurityPolicy: "Basic256Sha256",
+			SecurityMode:   "Sign",
+			Certfile:       "/home/wei/ca/clientcert.pem",
+			Keyfile:        "/home/wei/ca/clientkey.pem",
+			RemoteCertfile: "/home/wei/ca/servercert.pem",
+		*/
```
Comment (fisherxu) [inline pkg/opcua/driver/client_test.go:44]: We can remove it if unnecessary

**Review by fisherxu — COMMENTED** [review]
Thanks for the new Mapper, and please remove the binary :)

File: Makefile (hunk 2 above)
Comment (sailorvii) [reply Makefile:19]: Done.

File: pkg/opcua/ca/pass (hunk 3 above)
Comment (sailorvii) [reply pkg/opcua/ca/pass:1]: It's the password file. The original design is that: define the username & password in the device instance yaml and pass it by the configmap. I think it's dangerous and change to use the password file which is in local directory.

File: pkg/opcua/driver/client_test.go (hunk 4 above)
Comment (sailorvii) [reply pkg/opcua/driver/client_test.go:44]: Sorry, I'll refine the test. There're several test cases in one test function.

File: pkg/opcua/Dockerfile (hunk 1 above)
Comment (sailorvii) [reply pkg/opcua/Dockerfile:7]: The customer could package it without any file or with the files for them own. It depends on the authentication method the opcua server requires. This folder is an example that the customer could refer to.

**Comment by fisherxu** [comment]
@sailorvii  Let's fix any issues in follow-up PRs.

**Pull request merged**